### ISSUE TITLE
objects.inv ValueError

### ISSUE THREAD

**Issue by thijstriemstra** [sphinx-doc/sphinx#3488]
Subject: `ValueError` for `luma.core` package.

### Problem
It seems that Sphinx generates a corrupted `objects.inv` file for the https://github.com/rm-hull/luma.core project.

#### Procedure to reproduce the problem
- checkout the code and install the package:
```
git clone https://github.com/rm-hull/luma.core.git
cd luma.core
pip install -e .
```
go to the doc directory and build the project.

```
cd doc
make html
```
#### Error logs / results
The generated `objects.inv` throws an error for intersphinx in other projects:

```
WARNING: intersphinx inventory 'https://luma-core.readthedocs.io/en/latest/objects.inv' not readable due to ValueError: unknown or unsupported inventory version: ValueError()
```

Inspecting the generated `objects.inv` also throws the same error:
```
$ python -m sphinx.ext.intersphinx _build/html/objects.inv 
intersphinx inventory '_build/html/objects.inv' not readable due to ValueError: unknown or unsupported inventory version: ValueError()
```

When I add a print on [line 127](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/ext/intersphinx.py#L127) it shows something with zlib and also the bare value error:

```python
if 'zlib' not in line:
    print('zlib')
    raise ValueError
```

~~And downloading the `objects.inv` from readthedocs also has this problem:~~ **update**: file is fixed now, see comment below.

```
$ wget https://luma-core.readthedocs.io/en/latest/objects.inv
$ python -m sphinx.ext.intersphinx objects.inv
intersphinx inventory 'objects.inv' not readable due to ValueError: unknown or unsupported inventory version: ValueError()
````

#### Expected results
I expect the error to be in our project somewhere but IMO it should never throw this `ValueError`, or at least something more descriptive.

### Environment info
- OS: Ubuntu 16.04
- Python version: 3.5.3
- Sphinx version: 1.4.8, 1.5.0, 1.5.1, 1.5.2, 1.5.3

**Comment by thijstriemstra** [comment]
Commenting out the following option:

```diff
 # The encoding of source files.
 #
-# source_encoding = 'utf-8-sig'
+source_encoding = 'utf-8-sig'
```

Introduces some new prints during build:

```
# The remainder of this file is compressed using zlib.
```

But the resulting .inv file still throws a `ValueError`.

**Comment by rm-hull** [comment]
The header part shows as:

```python
# Sphinx inventory version 2
# Project: Luma.Core: SBC display drivers
# Version: 0.4.2

# The remainder of this file is compressed using zlib.
```

If you edit this to get rid of the newline so it reads

```diff
# Sphinx inventory version 2
# Project: Luma.Core: SBC display drivers
# Version: 0.4.2
# The remainder of this file is compressed using zlib.
```

Then the following output is produced:

```
$ python3 -m sphinx.ext.intersphinx objects.inv
py:attribute
	luma.core.const.common.DISPLAYALLON      api-documentation.html#luma.core.const.common.DISPLAYALLON
	luma.core.const.common.DISPLAYALLON_RESUME api-documentation.html#luma.core.const.common.DISPLAYALLON_RESUME
	luma.core.const.common.DISPLAYOFF        api-documentation.html#luma.core.const.common.DISPLAYOFF
	luma.core.const.common.DISPLAYON         api-documentation.html#luma.core.const.common.DISPLAYON
	luma.core.const.common.INVERTDISPLAY     api-documentation.html#luma.core.const.common.INVERTDISPLAY
	luma.core.const.common.NORMALDISPLAY     api-documentation.html#luma.core.const.common.NORMALDISPLAY
	luma.core.const.common.SETCONTRAST       api-documentation.html#luma.core.const.common.SETCONTRAST
	luma.core.const.common.SETMULTIPLEX      api-documentation.html#luma.core.const.common.SETMULTIPLEX
	luma.core.const.common.SETREMAP          api-documentation.html#luma.core.const.common.SETREMAP
py:class
	luma.core.const.common                   api-documentation.html#luma.core.const.common
	luma.core.device.device                  api-documentation.html#luma.core.device.device
	luma.core.device.dummy                   api-documentation.html#luma.core.device.dummy
	luma.core.framebuffer.diff_to_previous   api-documentation.html#luma.core.framebuffer.diff_to_previous
	luma.core.framebuffer.full_frame         api-documentation.html#luma.core.framebuffer.full_frame
	luma.core.legacy.font.proportional       api-documentation.html#luma.core.legacy.font.proportional
	luma.core.mixin.capabilities             api-documentation.html#luma.core.mixin.capabilities
	luma.core.render.canvas                  api-documentation.html#luma.core.render.canvas
	luma.core.serial.i2c                     api-documentation.html#luma.core.serial.i2c
	luma.core.serial.noop                    api-documentation.html#luma.core.serial.noop
	luma.core.serial.spi                     api-documentation.html#luma.core.serial.spi
	luma.core.sprite_system.dict_wrapper     api-documentation.html#luma.core.sprite_system.dict_wrapper
	luma.core.sprite_system.framerate_regulator api-documentation.html#luma.core.sprite_system.framerate_regulator
	luma.core.sprite_system.spritesheet      api-documentation.html#luma.core.sprite_system.spritesheet
	luma.core.threadpool.threadpool          api-documentation.html#luma.core.threadpool.threadpool
	luma.core.threadpool.worker              api-documentation.html#luma.core.threadpool.worker
	luma.core.virtual.history                api-documentation.html#luma.core.virtual.history
	luma.core.virtual.hotspot                api-documentation.html#luma.core.virtual.hotspot
	luma.core.virtual.snapshot               api-documentation.html#luma.core.virtual.snapshot
	luma.core.virtual.terminal               api-documentation.html#luma.core.virtual.terminal
	luma.core.virtual.viewport               api-documentation.html#luma.core.virtual.viewport
py:exception
	luma.core.error.DeviceAddressError       api-documentation.html#luma.core.error.DeviceAddressError
	luma.core.error.DeviceDisplayModeError   api-documentation.html#luma.core.error.DeviceDisplayModeError
	luma.core.error.DeviceNotFoundError      api-documentation.html#luma.core.error.DeviceNotFoundError
	luma.core.error.DevicePermissionError    api-documentation.html#luma.core.error.DevicePermissionError
	luma.core.error.Error                    api-documentation.html#luma.core.error.Error
py:function
	luma.core.legacy.show_message            api-documentation.html#luma.core.legacy.show_message
	luma.core.legacy.text                    api-documentation.html#luma.core.legacy.text
	luma.core.legacy.textsize                api-documentation.html#luma.core.legacy.textsize
	luma.core.virtual.calc_bounds            api-documentation.html#luma.core.virtual.calc_bounds
	luma.core.virtual.range_overlap          api-documentation.html#luma.core.virtual.range_overlap
py:method
	luma.core.device.device.capabilities     api-documentation.html#luma.core.device.device.capabilities
	luma.core.device.device.cleanup          api-documentation.html#luma.core.device.device.cleanup
	luma.core.device.device.clear            api-documentation.html#luma.core.device.device.clear
	luma.core.device.device.command          api-documentation.html#luma.core.device.device.command
	luma.core.device.device.contrast         api-documentation.html#luma.core.device.device.contrast
	luma.core.device.device.data             api-documentation.html#luma.core.device.device.data
	luma.core.device.device.display          api-documentation.html#luma.core.device.device.display
	luma.core.device.device.hide             api-documentation.html#luma.core.device.device.hide
	luma.core.device.device.preprocess       api-documentation.html#luma.core.device.device.preprocess
	luma.core.device.device.show             api-documentation.html#luma.core.device.device.show
	luma.core.device.dummy.capabilities      api-documentation.html#luma.core.device.dummy.capabilities
	luma.core.device.dummy.cleanup           api-documentation.html#luma.core.device.dummy.cleanup
	luma.core.device.dummy.clear             api-documentation.html#luma.core.device.dummy.clear
	luma.core.device.dummy.command           api-documentation.html#luma.core.device.dummy.command
	luma.core.device.dummy.contrast          api-documentation.html#luma.core.device.dummy.contrast
	luma.core.device.dummy.data              api-documentation.html#luma.core.device.dummy.data
	luma.core.device.dummy.display           api-documentation.html#luma.core.device.dummy.display
	luma.core.device.dummy.hide              api-documentation.html#luma.core.device.dummy.hide
	luma.core.device.dummy.preprocess        api-documentation.html#luma.core.device.dummy.preprocess
	luma.core.device.dummy.show              api-documentation.html#luma.core.device.dummy.show
	luma.core.framebuffer.diff_to_previous.getdata api-documentation.html#luma.core.framebuffer.diff_to_previous.getdata
	luma.core.framebuffer.diff_to_previous.inflate_bbox api-documentation.html#luma.core.framebuffer.diff_to_previous.inflate_bbox
	luma.core.framebuffer.diff_to_previous.redraw_required api-documentation.html#luma.core.framebuffer.diff_to_previous.redraw_required
	luma.core.framebuffer.full_frame.getdata api-documentation.html#luma.core.framebuffer.full_frame.getdata
	luma.core.framebuffer.full_frame.inflate_bbox api-documentation.html#luma.core.framebuffer.full_frame.inflate_bbox
	luma.core.framebuffer.full_frame.redraw_required api-documentation.html#luma.core.framebuffer.full_frame.redraw_required
	luma.core.mixin.capabilities.capabilities api-documentation.html#luma.core.mixin.capabilities.capabilities
	luma.core.mixin.capabilities.clear       api-documentation.html#luma.core.mixin.capabilities.clear
	luma.core.mixin.capabilities.display     api-documentation.html#luma.core.mixin.capabilities.display
	luma.core.mixin.capabilities.preprocess  api-documentation.html#luma.core.mixin.capabilities.preprocess
	luma.core.serial.i2c.cleanup             api-documentation.html#luma.core.serial.i2c.cleanup
	luma.core.serial.i2c.command             api-documentation.html#luma.core.serial.i2c.command
	luma.core.serial.i2c.data                api-documentation.html#luma.core.serial.i2c.data
	luma.core.serial.spi.cleanup             api-documentation.html#luma.core.serial.spi.cleanup
	luma.core.serial.spi.command             api-documentation.html#luma.core.serial.spi.command
	luma.core.serial.spi.data                api-documentation.html#luma.core.serial.spi.data
	luma.core.sprite_system.framerate_regulator.effective_FPS api-documentation.html#luma.core.sprite_system.framerate_regulator.effective_FPS
	luma.core.sprite_system.framerate_regulator.sleep api-documentation.html#luma.core.sprite_system.framerate_regulator.sleep
	luma.core.sprite_system.spritesheet.animate api-documentation.html#luma.core.sprite_system.spritesheet.animate
	luma.core.threadpool.threadpool.add_task api-documentation.html#luma.core.threadpool.threadpool.add_task
	luma.core.threadpool.threadpool.wait_completion api-documentation.html#luma.core.threadpool.threadpool.wait_completion
	luma.core.threadpool.worker.run          api-documentation.html#luma.core.threadpool.worker.run
	luma.core.virtual.history.display        api-documentation.html#luma.core.virtual.history.display
	luma.core.virtual.history.restore        api-documentation.html#luma.core.virtual.history.restore
	luma.core.virtual.history.savepoint      api-documentation.html#luma.core.virtual.history.savepoint
	luma.core.virtual.hotspot.paste_into     api-documentation.html#luma.core.virtual.hotspot.paste_into
	luma.core.virtual.hotspot.should_redraw  api-documentation.html#luma.core.virtual.hotspot.should_redraw
	luma.core.virtual.hotspot.update         api-documentation.html#luma.core.virtual.hotspot.update
	luma.core.virtual.snapshot.paste_into    api-documentation.html#luma.core.virtual.snapshot.paste_into
	luma.core.virtual.snapshot.should_redraw api-documentation.html#luma.core.virtual.snapshot.should_redraw
	luma.core.virtual.terminal.backspace     api-documentation.html#luma.core.virtual.terminal.backspace
	luma.core.virtual.terminal.carriage_return api-documentation.html#luma.core.virtual.terminal.carriage_return
	luma.core.virtual.terminal.clear         api-documentation.html#luma.core.virtual.terminal.clear
	luma.core.virtual.terminal.erase         api-documentation.html#luma.core.virtual.terminal.erase
	luma.core.virtual.terminal.flush         api-documentation.html#luma.core.virtual.terminal.flush
	luma.core.virtual.terminal.newline       api-documentation.html#luma.core.virtual.terminal.newline
	luma.core.virtual.terminal.println       api-documentation.html#luma.core.virtual.terminal.println
	luma.core.virtual.terminal.putch         api-documentation.html#luma.core.virtual.terminal.putch
	luma.core.virtual.terminal.puts          api-documentation.html#luma.core.virtual.terminal.puts
	luma.core.virtual.terminal.tab           api-documentation.html#luma.core.virtual.terminal.tab
	luma.core.virtual.viewport.add_hotspot   api-documentation.html#luma.core.virtual.viewport.add_hotspot
	luma.core.virtual.viewport.display       api-documentation.html#luma.core.virtual.viewport.display
	luma.core.virtual.viewport.is_overlapping_viewport api-documentation.html#luma.core.virtual.viewport.is_overlapping_viewport
	luma.core.virtual.viewport.refresh       api-documentation.html#luma.core.virtual.viewport.refresh
	luma.core.virtual.viewport.remove_hotspot api-documentation.html#luma.core.virtual.viewport.remove_hotspot
	luma.core.virtual.viewport.set_position  api-documentation.html#luma.core.virtual.viewport.set_position
py:module
	luma.core.const                          api-documentation.html#module-luma.core.const
	luma.core.device                         api-documentation.html#module-luma.core.device
	luma.core.error                          api-documentation.html#module-luma.core.error
	luma.core.framebuffer                    api-documentation.html#module-luma.core.framebuffer
	luma.core.legacy                         api-documentation.html#module-luma.core.legacy
	luma.core.legacy.font                    api-documentation.html#module-luma.core.legacy.font
	luma.core.mixin                          api-documentation.html#module-luma.core.mixin
	luma.core.render                         api-documentation.html#module-luma.core.render
	luma.core.serial                         api-documentation.html#module-luma.core.serial
	luma.core.sprite_system                  api-documentation.html#module-luma.core.sprite_system
	luma.core.threadpool                     api-documentation.html#module-luma.core.threadpool
	luma.core.virtual                        api-documentation.html#module-luma.core.virtual
std:doc
	api-documentation                        API Documentation                       : api-documentation.html
	index                                    Luma.Core                               : index.html
	install                                  Installation                            : install.html
	intro                                    Introduction                            : intro.html
std:label
	api documentation                        API Documentation                       : api-documentation.html#api-documentation
	arch linux                               Arch Linux                              : install.html#arch-linux
	changelog                                ChangeLog                               : index.html#changelog
	contributing                             Contributing                            : index.html#contributing
	contributors                             Contributors                            : index.html#contributors
	genindex                                 Index                                   : genindex.html
	github                                   GitHub                                  : index.html#github
	installation                             Installation                            : install.html#installation
	installing from pypi                     Installing from PyPi                    : install.html#installing-from-pypi
	installing from source                   Installing from source                  : install.html#installing-from-source
	introduction                             Introduction                            : intro.html#introduction
	luma.core                                Luma.Core                               : index.html#luma-core
	luma.core.const                          api-documentation.html#module-luma.core.const
	luma.core.device                         api-documentation.html#module-luma.core.device
	luma.core.error                          api-documentation.html#module-luma.core.error
	luma.core.framebuffer                    api-documentation.html#module-luma.core.framebuffer
	luma.core.legacy                         api-documentation.html#module-luma.core.legacy
	luma.core.legacy.font                    api-documentation.html#module-luma.core.legacy.font
	luma.core.mixin                          api-documentation.html#module-luma.core.mixin
	luma.core.render                         api-documentation.html#module-luma.core.render
	luma.core.serial                         api-documentation.html#module-luma.core.serial
	luma.core.sprite_system                  api-documentation.html#module-luma.core.sprite_system
	luma.core.threadpool                     api-documentation.html#module-luma.core.threadpool
	luma.core.virtual                        api-documentation.html#module-luma.core.virtual
	modindex                                 Module Index                            : py-modindex.html
	raspbian                                 Raspbian                                : install.html#raspbian
	search                                   Search Page                             : search.html
	the mit license (mit)                    The MIT License (MIT)                   : index.html#the-mit-license-mit
```

This can be tracked down to the version in [luma.core](https://github.com/rm-hull/luma.core) having a newline in it - this should now be squashed in https://github.com/rm-hull/luma.core/commit/119ddf9822e6293b2d1c326c3c6610d7d659cc17

**Comment by thijstriemstra** [comment]
thanks for the research @rm-hull!

@tk0miya this is something that should be fixed in Sphinx no?

**Comment by tk0miya** [comment]
Thank you for quite detail investigation!
Fixed at 3a3cec7.
